common/src/main/java/com/viaversion/viaversion/protocols/protocol1_13to1_12_2/storage/BlockConnectionStorage.java

```diff
@@ -88,10 +94,23 @@ public void remove(int x, int y, int z) {
         }
     }
 
+    public void markModified(Position pos) {
+        this.modified.add(pos);
+    }
+
+    public boolean recentlyModified(Position pos) {
```

Comment:
Using a global queue which never has less than 5 elements in it is a very big performance issue. Block connections by themself are already pretty slow, but adding at least 5 (but more likely even more) additional iterations to each block (inside a chunk) being checked/stored is way too much for solving such a small issue

Reply:
The performance on block connections is especially terrible for connecting neighbours across chunks, but that shouldnt be using this at all.

The affected performance here will be:
 - for every modified block
 - if the block is next to a fence/whetever
 - if the fence, on the client-side, has exactly the same shape as we've resolves it should have
 - then we check a list of at most 5 blocks

So no, not each single block is getting the 5 extra iterations, only the edge-case ones

That's also the reason i wanted to keep the list small, even if it has the potential for a player to place 10 blocks within ping period and there being a potential for a misconnected block.

Reply:
Yeah right I forgot Via already only checks against blocks which could have connections. It shouldn't be that bad then